pkg/resource/deploy/deployment_executor.go

```diff
@@ -159,23 +161,9 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 		ex.printPendingOperationsWarning()
 	}
 
-	// The set of -t targets provided on the command line.  'nil' means 'update everything'.
-	// Non-nil means 'update only in this set'.  We don't error if the user specifies a target
-	// during `update` that we don't know about because it might be the urn for a resource they
-	// want to create.
-	updateTargetsOpt := opts.UpdateTargets
-	replaceTargetsOpt := opts.ReplaceTargets
-	destroyTargetsOpt := opts.DestroyTargets
-	if res := ex.checkTargets(opts.ReplaceTargets, OpReplace); res != nil {
+	if res := ex.checkTargets(opts.ReplaceTargets); res != nil {
 		return nil, res
 	}
-	if res := ex.checkTargets(opts.DestroyTargets, OpDelete); res != nil {
-		return nil, res
-	}
-
-	if (updateTargetsOpt.IsConstrained() || replaceTargetsOpt.IsConstrained()) && destroyTargetsOpt.IsConstrained() {
-		contract.Failf("Should not be possible to have both .DestroyTargets and .UpdateTargets or .ReplaceTargets")
-	}
```

Comment:
UpdateTargets and DestroyTargets are now the same and `ex.checkTargets(opts.Target)` will fail if a non-existent resource is specified in Target.

The current approach pushes checking that `--target` values exist on `Destroy` earlier in `cmd/destroy.go` similar to the `--exclude-protected` flag.

Reply:
See comment in destroy.go about thoughts about moving that check back to here but in a toggled way for just Destroy and Refresh.